### PR TITLE
Add `include_seconds` option for `time_field`

### PR DESCRIPTION
**Pull request by ghiculescu**
Per https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time#time_value_format, if you format your time without seconds, browsers may render the input differerently. This adds support for this to Rails so that users don't need to call `strftime` on the value being passed into the form helper.

For example, in Chrome `<input value="16:22" type="time" />` renders this:

image

Whereas `<input value="16:22:01.440" type="time" />` (which is what `f.time_field` currently outputs) renders:

image

So with this PR, if you want the UI from the first screenshot, you could do this:

```ruby
form.time_field :foo, include_seconds: false
```

Rather than:

```ruby
form.time_field :foo, value: form.object.foo&.strftime("%H:%M")
```

### CODE REVIEW

File: actionview/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+*   Add `include_seconds` option for `time_field`
+
+        <%= form.time_field :foo, include_seconds: false %>
+        <%# => <input value="16:22" type="time" />
```
Comment (henrik) [inline actionview/CHANGELOG.md:4]:
```suggestion
        <%# => <input value="16:22" type="time" /> %>
```

File: actionview/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+
+    Default includes seconds:
+
+        <%= form.time_field :foo %>
+        <%# => <input value="16:22:01.440" type="time" />
```
Comment (henrik) [inline actionview/CHANGELOG.md:9]:
```suggestion
        <%# => <input value="16:22:01.440" type="time" /> %>
```

File: actionview/lib/action_view/helpers/form_helper.rb [hunk 3]
```diff
@@ -1424,6 +1425,12 @@ def date_field(object_name, method, options = {})
       #   time_field("task", "started_at", min: "01:00:00")
       #   # => <input id="task_started_at" name="task[started_at]" type="time" min="01:00:00.000" />
       #
+      # By default, provided times will be formatted including seconds. You can just render the hour
```
Comment (henrik) [inline actionview/lib/action_view/helpers/form_helper.rb:1428]:
```suggestion
      # By default, provided times will be formatted including seconds. You can render just the hour
```

**Review by henrik — COMMENTED** [review]
🥳

File: actionview/CHANGELOG.md (hunk 1 above)
Comment (ghiculescu) [reply actionview/CHANGELOG.md:4]: I think `# => <input value="16:22" type="time" />` might actually be correct. Will fix this + the entry I copied the bad syntax from.

Comment (henrik) [reply actionview/CHANGELOG.md]: Yeah, that would read even better, I think 👍

File: actionview/lib/action_view/helpers/tags/time_field.rb [hunk 4]
```diff
@@ -4,9 +4,19 @@ module ActionView
   module Helpers
     module Tags # :nodoc:
       class TimeField < DatetimeField # :nodoc:
+        def render
+          @include_seconds = @options.delete(:include_seconds)
```
Comment (rafaelfranca) [inline actionview/lib/action_view/helpers/tags/time_field.rb:8]: This should be set in the initialize, not in the render method.

File: actionview/lib/action_view/helpers/tags/time_field.rb [hunk 5]
```diff
@@ -4,9 +4,19 @@ module ActionView
...
+          @include_seconds = true if @include_seconds.nil?
```
Comment (rafaelfranca) [inline actionview/lib/action_view/helpers/tags/time_field.rb:9]:
```suggestion
          @include_seconds = @options.delete(:include_seconds { true }
```

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

**Comment by ghiculescu** [comment]
@rafaelfranca thanks! I've tidied up based on those suggestions.

**Pull request merged**